sys-apps/ignition/files/0005-config-v3_4-convert-ignition-2.x-to-3.4-exp.patch

```diff
@@ -0,0 +1,81 @@
+From 85d70f5dab2014859a1248ced91271d67a813714 Mon Sep 17 00:00:00 2001
+From: Mathieu Tortuyaux <mtortuyaux@microsoft.com>
+Date: Wed, 25 May 2022 10:20:09 +0200
+Subject: [PATCH 05/15] config/v3_4: convert ignition 2.x to 3.4-exp
+
+if the version of ignition is 2.x we convert it to 3.4-exp using
+ign-converter.
+
+it shoud support any 2.x version (or at the least the last 2 releases)
+
+Signed-off-by: Mathieu Tortuyaux <mtortuyaux@microsoft.com>
+---
+ config/v3_4_experimental/config.go | 38 ++++++++++++++++++++++++++++++
+ 1 file changed, 38 insertions(+)
+
+diff --git a/config/v3_4_experimental/config.go b/config/v3_4_experimental/config.go
+index 512fc9f0..b0375236 100644
+--- a/config/v3_4_experimental/config.go
++++ b/config/v3_4_experimental/config.go
+@@ -15,6 +15,10 @@
+ package v3_4_experimental
+ 
+ import (
++	"encoding/json"
++	"fmt"
++
++	"github.com/flatcar-linux/ignition/config/v2_4"
+ 	"github.com/flatcar-linux/ignition/v2/config/merge"
+ 	"github.com/flatcar-linux/ignition/v2/config/shared/errors"
+ 	"github.com/flatcar-linux/ignition/v2/config/util"
+@@ -25,6 +29,8 @@ import (
+ 
+ 	"github.com/coreos/go-semver/semver"
+ 	"github.com/coreos/vcontext/report"
++
++	"github.com/flatcar-linux/ign-converter/translate/v24tov31"
+ )
+ 
+ func Merge(parent, child types.Config) types.Config {
+@@ -67,6 +73,38 @@ func ParseCompatibleVersion(raw []byte) (types.Config, report.Report, error) {
+ 		return types.Config{}, rpt, err
+ 	}
+ 
++	// if the version is 2.x or 1.x, we
++	// convert it to 3.1
++	if version.Major != 3 {
++		// Parse should fallback on every 2.x supported version
++		cfg, _, err := v2_4.Parse(raw)
++		if err != nil || rpt.IsFatal() {
++			return types.Config{}, report.Report{}, fmt.Errorf("unable to parse 2.x ignition: %w", err)
++		}
++
++		/*
++			map[string]string{} is required by the ign-converter
++			Ignition Spec 3 will mount filesystems at the mountpoint specified by path when running.
++			Filesystems no longer have the name field and files, links, and directories no longer specify the filesystem by name.
++			This means to translate filesystems (with the exception of root),
++			you must also provide a mapping of filesystem name to absolute path, e.g.
++			```
++			map[string]string{"var": "/var"}
++			```
++		*/
++		newCfg, err := v24tov31.Translate(cfg, map[string]string{})
```

Comment:
This comment is confusing me, because it talks about why the map is required, only to pass an empty one. Who should be responsible for filling the map? Is passing an empty map going to make the conversion to fail, because some required entry was missing from the map?

Reply:
I get your point - this is required by `ign-converter` in the `Translate` method. Later in the code, the map is feed, for example `"root": "/"` is automatically added.
We also added some logic to generate values see this commit: https://github.com/flatcar-linux/ign-converter/commit/96f82d676ffd35e6dbf68078455ff02affd7b8b1.
If you want more information on that, there is the upstream issue: https://github.com/coreos/ign-converter/issues/30.

When `ign-converter` is used via its CLI, `fsMap` is manually provided by the user.